crates/next-core/src/next_client/context.rs

```diff
@@ -350,6 +350,9 @@ pub async fn get_client_module_options_context(
         enable_postcss_transform,
         side_effect_free_packages: next_config.optimize_package_imports().owned().await?,
         keep_last_successful_parse: next_mode.is_development(),
+        remove_unused_exports: *next_config
+            .turbopack_remove_unused_exports(next_mode.is_development())
+            .await?,
```

Comment:
Could we add that to the `tree_shaking_mode`:

```rs
pub enum TreeShakingMode {
    ModuleFragments,
    UnusedExportsRemoval,
    ReexportsOnly,
}
```

Reply:
That's how I tried to implement initially, but this is global and we cannot use different tree shaking mode for user code and foreign code (`node_modules`).

Initially I added it to `TreeShakingMode` and did 

```rust
enable_unused_export_removal: matches!(user_tree_shaking_mode, TreeShakingMode::UnusedExportsRemoval) || matches!(foreign_tree_shaking_mode, TreeShakingMode::UnusedExportsRemoval),
```

<img width="816" alt="image" src="https://github.com/user-attachments/assets/e2af009b-45e7-4297-adb4-697d036c2ccf" />

But the code sounds like I should rather add a bool flag, so I migrated it to a bool config.